tests/diagnostic/test_main.cpp

```diff
@@ -184,51 +160,97 @@ TYPED_TEST(SimulatorTest, particles)
 
-    ParticlesHi5Diagnostic<Hierarchy, HybridModel> hi5{hierarchy, hybridModel,
-                                                       HighFive::File::ReadOnly};
-    auto& writer = hi5.writer;
+    { // scoped to destruct after dump
+        Hi5Diagnostic<Hierarchy, HybridModel> hi5{hierarchy, hybridModel, NEW_HI5_FILE};
+        hi5.dMan.addDiagDict(hi5.particles("/ions/pop/ions_alpha/domain"))
+            .addDiagDict(hi5.particles("/ions/pop/ions_alpha/levelGhost"))
+            .addDiagDict(hi5.particles("/ions/pop/ions_alpha/patchGhost"))
+            .addDiagDict(hi5.particles("/ions/pop/ions_protons/domain"))
+            .addDiagDict(hi5.particles("/ions/pop/ions_protons/levelGhost"))
+            .addDiagDict(hi5.particles("/ions/pop/ions_protons/patchGhost"))
+            .dump();
+    }
+
+    Hi5Diagnostic<Hierarchy, HybridModel> hi5{hierarchy, hybridModel, HighFive::File::ReadOnly};
+    validateParticleDump(sim, hi5.writer);
+}
+
+template<typename Simulator, typename Writer>
+void validateAttributes(Simulator& sim, Writer& hi5)
+{
+    using GridLayout = typename Simulator::PHARETypes::GridLayout_t;
+
+    auto& hybridModel = *sim.getHybridModel();
+    auto hifile       = hi5.makeFile(hi5.fileString("/EM_B")); // all files have attributes
 
     auto visit = [&](GridLayout& gridLayout, std::string patchID, size_t iLevel) {
-        std::string patchPath = writer.getPatchPath("time", iLevel, patchID) + "/", origin;
-        writer.file().getGroup(patchPath).getAttribute("origin").read(origin);
+        std::string patchPath = hi5.getPatchPath("time", iLevel, patchID), origin;
+        hifile->file().getGroup(patchPath).getAttribute("origin").read(origin);
         EXPECT_EQ(gridLayout.origin().str(), origin);
     };
 
-    PHARE::amr::visitHierarchy<GridLayout>(hierarchy, *hybridModel.resourcesManager, visit, 0,
+    PHARE::amr::visitHierarchy<GridLayout>(*sim.getPrivateHierarchy(),
+                                           *hybridModel.resourcesManager, visit, 0,
                                            sim.getNumberOfLevels(), hybridModel);
-    // sim.visitHierarchy(visit, hybridModel);
+}
+
+
+TYPED_TEST(SimulatorTest, allFromPython)
+{
+    using HybridModel = typename TypeParam::HybridModel;
+    using Hierarchy   = typename TypeParam::Hierarchy;
+
+    TypeParam sim;
+    sim.dMan->dump();
+    sim.dMan.reset();
+    sim.modelView.reset();
+    sim.writer.reset(); // keeps a handle and breaks the following readonly opening
+
+    auto& hybridModel = *sim.getHybridModel();
+    auto& hierarchy   = *sim.getPrivateHierarchy();
+
+    Hi5Diagnostic<Hierarchy, HybridModel> hi5{hierarchy, hybridModel, HighFive::File::ReadOnly};
+
+    validateFluidDump(sim, hi5.writer);
+    validateElectromagDump(sim, hi5.writer);
+    validateParticleDump(sim, hi5.writer);
+    validateAttributes(sim, hi5.writer);
 }
 
 int main(int argc, char** argv)
 {
+    kul::Dir{"phare_outputs"}.mk();
```

Comment:
highfive doesn't appear to make directories to files if the path give to HighFive::File includes them

Reply:
oh! Yes not surprised. However I would prefer that if the user specifies directory paths for outputs, that those are checked to exist from python before running PHARE. I don't want the user to wait 10hours for the job to start and finally get an error because the path is invalid and the directory can't be made

Reply:
in the normal case we do or don't create directories then?

Reply:
I would say phare expect all directories to exist so if something creates them it's python

Reply:
done in the simulation.py constructor/validation step